plugin/templates/resource_name.py

```diff
@@ -188,3 +289,54 @@ def __init__(self, fixed_config, java_package, oneof):
 
     def template_name(self):
         return "resource_name_fixed.mustache"
+
+
+def get_id_segments(pattern):
+    name_template = path_template.PathTemplate(pattern)
+    return [
+        seg.literal for seg in name_template.segments
+        if seg.kind == path_template._BINDING
+    ]
+
+
+def get_format_field(lower_underscore, symbol):
+    return {
+        'lower_underscore': lower_underscore,
+        'lower_camel': casing_utils.lower_underscore_to_lower_camel(
+                       lower_underscore),
+        'lower_camel_symbol': symbol,
+        'upper_underscore': casing_utils.lower_underscore_to_upper_underscore(
+                            lower_underscore),
+        'upper_camel': casing_utils.lower_underscore_to_upper_camel(
+                       lower_underscore),
+        'not_first': True,
+        'not_last': True
+    }
+
+
+def get_format_fields_for_pattern(pattern,
+                                  pattern_to_id_segments,
+                                  segment_to_segment_symbols):
+    if pattern not in pattern_to_id_segments:
+        return []
+    format_fields = [get_format_field(seg, segment_to_segment_symbols[seg])
+                     for seg in pattern_to_id_segments[pattern]]
+    format_fields[0]['not_first'] = False
+    format_fields[-1]['not_last'] = False
+    return format_fields
+
+
+def is_fixed_pattern(pattern):
+    return not('{' in pattern or '*' in pattern)
+
+
+def get_pattern_name(pattern):
+    if is_fixed_pattern(pattern):
+        start_index = \
+            next(i for (i, c) in enumerate(list(pattern)) if c.isalpha())
+        end_index = len(pattern) - next(
+            i for (i, c) in enumerate(list(pattern)[::-1]) if c.isalpha())
+        name_parts = re.split(r'[^a-zA-Z]', pattern[start_index:end_index])
+        return '_'.join(name_parts)
+    else:
```

Comment:
nit: If you inverse the original if statement (L334) you can simply return this at the beginning of this method, and avoid having the extra indentation for the other (bigger part) of your method. It is subjective, but is inline with google code practices of avoiding unnecessary code indentation blocks (code looks simpler this way).